### ISSUE TITLE
Bug: Quoted strings rule doesn't detect unquoted strings in array

### ISSUE THREAD

**Issue by mmerfort** [adrienverge/yamllint#208]
Hello,

I noticed that yamllint does not detect unquoted strings in an array even though this rule has been enabled:

### testfile.yaml
```yaml
---
bla:
  - FOO
  - BAR
  - BAZ
```
### configuration.yaml
```yaml
---
extends: 'default'

rules:
  quoted-strings:
    quote-type: 'single'
```

I tried using different quote-types but none of them seem to work.

Thanks in advance.

**Comment by adrienverge** [comment]
Hi @mmerfort, thanks for the report, I can confirm the bug.

It also happens for:
```yaml
ble: [FOO, BAR, BAZ]
bli: {FOO, BAR, BAZ}
```

I tested older versions of yamllint, and it seems that this doesn't work since the first commit that added the rule.

Any contribution contribution to fix this (and to add a non-regression test) would be welcome! https://github.com/adrienverge/yamllint/blob/0f073f7/yamllint/rules/quoted_strings.py
